### PR TITLE
Add DefaultAzureCredential support to AzureContainerRegistryHook

### PR DESCRIPTION
**Pull request by Lee-W**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/container_registry.py [hunk 1]
```diff
@@ -40,6 +43,24 @@ class AzureContainerRegistryHook(BaseHook):
...
+        return {
+            "subscription_id": StringField(
+                lazy_gettext("Subscription ID (optional)"),
+                widget=BS3TextFieldWidget(),
+            ),
+            "resource_group": StringField(
+                lazy_gettext("Resource group name (optional)"),
+                widget=BS3TextFieldWidget(),
+            ),
+        }
```
Comment (Taragolis) [inline airflow/providers/microsoft/azure/hooks/container_registry.py:62]: Would be nice to have information about this extra fields into the documentation https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/docs/apache-airflow-providers-microsoft-azure/connections/acr.rst?plain=1#L44-L56

Comment (Lee-W) [reply airflow/providers/microsoft/azure/hooks/container_registry.py:62]: Sure! Just updated these 2 fields

**Review by pankajastro — APPROVED** [review]

File: airflow/providers/microsoft/azure/provider.yaml [hunk 2]
```diff
@@ -85,6 +84,7 @@ dependencies:
   #   See issue https://github.com/apache/airflow/issues/30199
   - azure-mgmt-containerinstance>=7.0.0,<9.0.0
   - azure-mgmt-datafactory>=1.0.0,<2.0
+  - azure-mgmt-containerregistry>=8.0.0
```
Comment (eladkal) [inline airflow/providers/microsoft/azure/provider.yaml]: please place this above line 83
It's not part of the TODO block

Comment (Lee-W) [reply airflow/providers/microsoft/azure/provider.yaml]: Sure! Just updated it

**Comment by Lee-W** [comment]
The static check seems to be canceled again.

**Comment by potiuk** [comment]
> The static check seems to be canceled again.

Seems like it took 12 minutes to just "start" installing the environment for mypy checks. So this is likely something flaky/environmental:

```
Tue, 29 Aug 2023 09:18:29 GMT
[INFO] This may take a few minutes...
Tue, 29 Aug 2023 09:30:52 GMT
[INFO] Installing environment for https://github.com/Lucas-C/pre-commit-hooks.
```


> This takes usually 20 seconds or so on public runners

```
Tue, 29 Aug 2023 08:59:55 GMT
[INFO] Installing environment for https://github.com/thlorenz/doctoc.git.
Tue, 29 Aug 2023 08:59:55 GMT
[INFO] Once installed this environment will be reused.
Tue, 29 Aug 2023 08:59:55 GMT
[INFO] This may take a few minutes...
Tue, 29 Aug 2023 09:00:15 GMT
[INFO] Installing environment for https://github.com/Lucas-C/pre-commit-hooks.
```

**Comment by potiuk** [comment]
Right - this is a "Real" problem now you see.

**Comment by Lee-W** [comment]
> Right - this is a "Real" problem now you see.

Got it. Just get it fixed. Thanks!

**Pull request merged**